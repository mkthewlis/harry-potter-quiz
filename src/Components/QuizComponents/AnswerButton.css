.answer-button {
    min-height: 1.8rem;
    min-width: 40rem;
    background-color: #FFFFFF;
    border-radius: 10px;
    align-items: center;
    cursor: pointer;
    border: 2px solid #FFC500;
    color: #7F0909;
    font-size: 1.5em;
    padding: .5rem;
    margin: 1rem 0;
    font-family: 'Courier New', Courier, monospace;
}