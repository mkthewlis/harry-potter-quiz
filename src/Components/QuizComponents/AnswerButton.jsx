import React from 'react';
import './AnswerButton.css';

export const AnswerButton = ({index, answer, ...onClick}) => (
    <>
        <button key={index} className="answer-button" {...onClick}>{answer}</button>
    </>
);