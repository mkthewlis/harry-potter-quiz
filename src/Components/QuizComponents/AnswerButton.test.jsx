import React from 'react';
import { shallow } from 'enzyme';
import { AnswerButton } from './AnswerButton';

describe('AnswerButton', () => {
    let wrapper;
    const onClickMock = jest.fn();

    beforeEach(() => {
        wrapper = shallow(
          <AnswerButton {...onClickMock}/>,
        );
      });
    it('renders without crashing', () => {
      expect(wrapper).toMatchSnapshot();
    });
});