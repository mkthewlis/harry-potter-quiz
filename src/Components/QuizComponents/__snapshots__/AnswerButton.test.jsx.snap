// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerButton renders without crashing 1`] = `
<Fragment>
  <button
    _isMockFunction={true}
    className="answer-button"
    getMockImplementation={[Function]}
    getMockName={[Function]}
    mock={
      Object {
        "calls": Array [],
        "instances": Array [],
        "invocationCallOrder": Array [],
        "results": Array [],
      }
    }
    mockClear={[Function]}
    mockImplementation={[Function]}
    mockImplementationOnce={[Function]}
    mockName={[Function]}
    mockRejectedValue={[Function]}
    mockRejectedValueOnce={[Function]}
    mockReset={[Function]}
    mockResolvedValue={[Function]}
    mockResolvedValueOnce={[Function]}
    mockRestore={[Function]}
    mockReturnThis={[Function]}
    mockReturnValue={[Function]}
    mockReturnValueOnce={[Function]}
  />
</Fragment>
`;
